package-res/resources/web/pentaho/type/Type.js

```diff
@@ -92,11 +92,17 @@ define([
       if(arg.optional(keyArgs, "isRoot"))
         O.setConst(this, "root", this);
 
+      // ----
       // Block inheritance, with default values
-      this._id         = null;
+
+      // Don't use inherited property definition which may be writable false
+      Object.defineProperty(this, "_id", {value: null, writable: true});
+
       this._shortId    = undefined;
       this._styleClass = null;
-      this._hasDescendants = false;
+
+      // Don't use inherited property definition which may be writable false
```

Comment:
Suggest changing to:

Don't use the inherited property definition, which may be writable as 'false'.